src/Control/Email/Email.php

```diff
@@ -753,215 +370,145 @@ public function getHTMLTemplate()
 
     /**
      * Set the template to render the email with
-     *
-     * @param string $template
-     * @return $this
      */
-    public function setHTMLTemplate($template)
+    public function setHTMLTemplate(string $template): static
     {
         if (substr($template ?? '', -3) == '.ss') {
             $template = substr($template ?? '', 0, -3);
         }
         $this->HTMLTemplate = $template;
-
         return $this;
     }
 
     /**
      * Get the template to render the plain part with
-     *
-     * @return string
      */
-    public function getPlainTemplate()
+    public function getPlainTemplate(): string
     {
         return $this->plainTemplate;
     }
 
     /**
      * Set the template to render the plain part with
-     *
-     * @param string $template
-     * @return $this
      */
-    public function setPlainTemplate($template)
+    public function setPlainTemplate(string $template): static
     {
         if (substr($template ?? '', -3) == '.ss') {
             $template = substr($template ?? '', 0, -3);
         }
         $this->plainTemplate = $template;
-
-        return $this;
-    }
-
-    /**
-     * @param array $recipients
-     * @return $this
-     */
-    public function setFailedRecipients($recipients)
-    {
-        $this->failedRecipients = $recipients;
-
         return $this;
     }
 
-    /**
-     * @return array
-     */
-    public function getFailedRecipients()
-    {
-        return $this->failedRecipients;
-    }
-
-    /**
-     * Used by {@link SSViewer} templates to detect if we're rendering an email template rather than a page template
-     *
-     * @return bool
-     */
-    public function IsEmail()
-    {
-        return true;
-    }
-
     /**
      * Send the message to the recipients
-     *
-     * @return bool true if successful or array of failed recipients
      */
-    public function send()
+    public function send(): void
     {
-        if (!$this->getBody()) {
-            $this->render();
-        }
-        if (!$this->hasPlainPart()) {
-            $this->generatePlainPartFromBody();
-        }
-        return Injector::inst()->get(Mailer::class)->send($this);
+        $this->render();
```

Comment:
If I call `$email->setBody($html);` or `$email->html($html);` to set the HTML body of the email, I would expect that body to be sent - won't calling `render()` here throw away the body I've set?

Reply:
No it won't, it will retain $html. There's a unit-testing covering this scenario e.g. everything that uses the `private function createTestEmail()`